src/harness/testcases/WINNF_FT_S_SIQ_testcase.py

```diff
@@ -841,6 +843,219 @@ def test_WINNF_FT_S_SIQ_11(self):
+
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_SIQ_12_default_config)
+  def test_WINNF_FT_S_SIQ_12(self, config_filename):
+    """[Configurable] Spectrum inquiry in the presence of protected entities."""
+
+    config = loadConfig(config_filename)
+
+    # Light checking of the config file
+    self.assertEqual(len(config['spectrumInquiryRequestsN2']),len(config['registrationRequestsN2']))
+    self.assertEqual(len(config['expectedResponseCodes']),len(config['spectrumInquiryRequestsN2']))
+    self.assertEqual(len(config['spectrumInquiryRequestsN2']),len(config['fr1Cbsd']))
+    self.assertEqual(len(config['spectrumInquiryRequestsN2']),len(config['fr2Cbsd']))
+    self.assertGreater(len(config['registrationRequestsN2']),0)
+
+    # Step1: Load information about N1 GWBZs
+    for gwpz_record in config['gwpzRecordsN1']:
+      self._sas_admin.InjectWisp(gwpz_record)
+
+    # Step2&3: Register N2 CBSDs
+    # Check registration response
+    # The assertRegistered function does the Inject FCC ID and user ID for the registration requests
+    cbsd_ids = self.assertRegistered(config['registrationRequestsN2'],config['conditionalParameters'])
+
+    # Step4: Load N3 PPA
+    for pal_record in config['palRecordsN3']:
+      self._sas_admin.InjectPalDatabaseRecord(pal_record[0])
+    for ppa_record in config['ppaRecordsN3']:
+      zone_id = self._sas_admin.InjectZoneData({"record": ppa_record})
+
+ 
+    # Step5: Trigger CPAS activity
+    if (len(config['gwpzRecordsN1']) > 0 or len(config['palRecordsN3']) > 0):
+      self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete() 
+
+    # Step6: Send N2 spectrum inquiry requests (one per registered CBSD)
+    spectrum_inquiry_list = config['spectrumInquiryRequestsN2']
+    addCbsdIdsToRequests(cbsd_ids, spectrum_inquiry_list)
+ 
+    spectrum_inquiry_request = {'spectrumInquiryRequest': spectrum_inquiry_list}
+    spectrum_inquiry_response = self._sas.SpectrumInquiry(spectrum_inquiry_request)['spectrumInquiryResponse']
+
+    # Check the response contains N2 SpectrumInquiryResponse objects
+    self.assertEqual(len(spectrum_inquiry_response),len(config['spectrumInquiryRequestsN2']))
+
+    for index,response in enumerate(spectrum_inquiry_response):
+      # If the corresponding request contained a valid cbsdId, the
+      # response shall contain the same cbsdId.
+      if 'cbsdId' in spectrum_inquiry_list[index]:
+        if spectrum_inquiry_list[index]['cbsdId'] in cbsd_ids:
+          self.assertEqual(response['cbsdId'], spectrum_inquiry_list[index]['cbsdId'])
+        else:
+          self.assertFalse('cbsdId' in response)
+
+      # Check response code matches the expected response code
+      self.assertIn(response['response']['responseCode'],config['expectedResponseCodes'][index])
+      if (response['response']['responseCode'] == 0) :
+        # Check if the 'availableChannel' parameter is present
+        self.assertTrue('availableChannel' in response)
```

Comment:
I'd suggest to insert here the else condition line ('availableChannel' is not present if the response code is not SUCCESS). So that this item of the test check list if more easy to read.

Even if we need to repeat the if condition to test the following items.

Reply:
I agree   that it will make it easier to read but it will also make it to deviate from   the test spec. An explanation has to be added on the reason for the   deviation.          I think it is better to leave it the way it is and aligned to the spec.